spec/rubocop/cop/sorbet/t_struct_property_attr_macro_spec.rb

```diff
@@ -0,0 +1,297 @@
+# frozen_string_literal: true
+
+RSpec.describe(RuboCop::Cop::Sorbet::TStructPropertyAttrMacro, :config) do
+  context("when using `const` property") do
+    context("and `attr_reader` is used") do
+      context("and the names match") do
+        it("registers an offense") do
+          expect_offense(<<~RUBY)
+            class Foo < T::Struct
+              attr_reader :bar
+              ^^^^^^^^^^^^^^^^ Do not override `T::Struct` property reader unless customizing it.
+
+              const :bar, String
+            end
+          RUBY
+        end
+
+        context("and the order is reversed") do
+          it("registers an offense") do
+            expect_offense(<<~RUBY)
+              class Foo < T::Struct
+                const :bar, String
+
+                attr_reader :bar
+                ^^^^^^^^^^^^^^^^ Do not override `T::Struct` property reader unless customizing it.
+              end
+            RUBY
+          end
+        end
+      end
+
+      context("and the names do not match") do
+        it("registers no offenses") do
+          expect_no_offenses(<<~RUBY)
+            class Foo < T::Struct
+              attr_reader :other
+              const :bar, String
+            end
+          RUBY
+        end
+      end
+    end
+
+    context("and a matching custom reader is defined without `attr_reader`") do
+      it("does not register an offense") do
+        expect_no_offenses(<<~RUBY)
+          class Foo < T::Struct
+            const :bar, String
+
+            def bar
+              # ...
+            end
+          end
+        RUBY
+      end
+    end
+
+    context("and `attr_writer` is used") do
+      context("and the names match") do
+        it("registers an offense") do
+          expect_offense(<<~RUBY)
+            class Foo < T::Struct
+              attr_writer :bar
```

Comment:
One case in which this might be legit would be if the writer's `sig` is **more** restrictive:

```ruby
class Foo < T::Struct
  sig { param(number: Integer) }
  attr_writer :number
  
  const :number, Numeric
end
```

_Less_ restrictive would be an error, although would be hard to check for.

This would be pretty unusual though, and arguably a better named method would probably make more sense, so it's probably safe to just assume that if `attr_*` is being used, it's incorrect usage, and let the author of such an exotic usage simply disable the cop.